azurerm/internal/services/web/static_site_custom_domain_resource.go

```diff
@@ -0,0 +1,218 @@
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/web/parse"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/tf/pluginsdk"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/timeouts"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+const (
+	txtValidationType   = "dns-txt-token"
+	cnameValidationType = "cname-delegation"
+)
+
+func resourceStaticSiteCustomDomain() *pluginsdk.Resource {
+	return &pluginsdk.Resource{
+		Create: resourceStaticSiteCustomDomainCreateOrUpdate,
+		Read:   resourceStaticSiteCustomDomainRead,
+		Delete: resourceStaticSiteCustomDomainDelete,
+		Importer: pluginsdk.ImporterValidatingResourceId(func(id string) error {
+			_, err := parse.StaticSiteCustomDomainID(id)
+			return err
+		}),
+
+		Timeouts: &pluginsdk.ResourceTimeout{
+			Create: pluginsdk.DefaultTimeout(30 * time.Minute),
+			Read:   pluginsdk.DefaultTimeout(5 * time.Minute),
+			Update: pluginsdk.DefaultTimeout(30 * time.Minute),
+			Delete: pluginsdk.DefaultTimeout(30 * time.Minute),
+		},
+
+		Schema: map[string]*pluginsdk.Schema{
+			"static_site_id": {
+				Type:         pluginsdk.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: azure.ValidateResourceID,
+			},
+
+			"name": {
+				Type:         pluginsdk.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringIsNotEmpty,
+			},
+
+			"validation_type": {
+				Type:     pluginsdk.TypeString,
+				Required: true,
+				ForceNew: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					txtValidationType,
+					cnameValidationType,
+				}, false),
+			},
+
+			"validation_token": {
+				Type:      pluginsdk.TypeString,
+				Sensitive: true,
+				Computed:  true,
+			},
+		},
+	}
+}
+
+func resourceStaticSiteCustomDomainCreateOrUpdate(d *pluginsdk.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).Web.StaticSitesClient
+	ctx, cancel := timeouts.ForCreate(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	log.Printf("[INFO] preparing arguments for AzureRM Static Site custom domain creation.")
+
+	staticSiteId, err := parse.StaticSiteID(d.Get("static_site_id").(string))
+
+	if err != nil {
+		return err
+	}
+
+	domainName := d.Get("name").(string)
+
+	resourceGroup := staticSiteId.ResourceGroup
+	staticSiteName := staticSiteId.Name
+
+	_, err = client.GetStaticSite(ctx, resourceGroup, staticSiteName)
+	if err != nil {
+		return fmt.Errorf("finding Static Site (Name %q / Resource Group %q): %s", staticSiteName, resourceGroup, err)
+	}
+
+	if d.IsNewResource() {
+		existing, err := client.GetStaticSiteCustomDomain(ctx, staticSiteId.ResourceGroup, staticSiteName, domainName)
+		if err != nil {
+			if !utils.ResponseWasNotFound(existing.Response) {
+				return fmt.Errorf("failed checking for presence of existing %s: %+v", staticSiteId, err)
```

Comment:
```suggestion
				return fmt.Errorf("checking for presence of existing %s: %+v", staticSiteId, err)
```
Nitpick semantics :)